signer/src/adapter/addns.c

```diff
@@ -581,7 +581,6 @@ dnsin_update(dnsin_type** addns, const char* filename, time_t* last_mod)
     } else {
         ods_log_error("[%s] unable to update dnsin: dnsin_read(%s) "
             "failed (%s)", adapter_str, filename, ods_status2str(status));
-        dnsin_cleanup(*addns);
     }
```

Comment:
Actually, now I'm reviewing I think instead of removing this line we should add:
*addns=NULL;
We should also do this in addns.c:691 in dnsout_update(). That way we prevent the crash *AND* not have the leak.